File: 2181. Merge Nodes in Between Zeros/2181. Merge Nodes in Between Zeros.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        count=0
        value=0
        result=ListNode(0)
        new=result
        while head:
            if head.val==0:
                count+=1
            else:
                value+=head.val
            if count==2:
                val=ListNode(value)
                new.next=val
                new=new.next
                value=0
                count-=1
            head=head.next
        return result.next
        

File: 2181. Merge Nodes in Between Zeros/2181. Merge Nodes in Between Zeros.cc
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeNodes(ListNode* head) {
        ListNode* ptr = head->next->next;
        ListNode* start = head->next;
        while (ptr) {
            int sum = 0;
            while (ptr && ptr->val != 0) {
                sum += ptr->val;
                start->next = ptr->next;
                delete ptr;
                ptr = start->next;
            }
            if (ptr) {
                start->val += sum;
                start->next = ptr->next;
                delete ptr;
                ptr = NULL;
                start = start->next;
                if (start)
                    ptr = start->next;
            }
        }
        head = head->next;
        return head;
    }
};